### DISCUSSION TITLE
Custom extension is triggering code from another extension

### DISCUSSION THREAD

**Discussion by MtDalPizzol** [awcodes/filament-tiptap-editor#296] [Q&A]
Hi, folks! I'm following the README to create a custom extension. This extension will do something similar to what Curator does: when you click the extension button in the toolbar, it will open a modal to select related articles to be embedded in the contents. In fact, I'm using Curator for my media and I'm using the code to base my code on what Curator already does.

The problem I'm having, is that when I click my custom extension button in the toolbar, it opens the modal I expect it to open, , but it ALSO runs the code from the Curator panel. I'm not sure if this is something expected for some internal LW update reason or if it's in fact a bug.

If you add the following around line 85 on `filament-curator/resources/views/components/modals/curator-panel.blade.php` you will get a `dd` when clicking the button for the custom extension, that has no relation whatsoever with curator.

```html
    <!-- Gallery -->
    @php
        dd('test');
    @endphp
```

And here's the code for the custom extension:

**/my-app/resources/views/components/tiptap/extensions/embedded-articles/embedded-articles-button.blade.php**

```blade
@props([
    'statePath' => null,
])

<x-filament-tiptap-editor::button label="Embed article" active="embeddedarticle" action="openArticlesModal()"
    x-data="{
        openArticlesModal() {
            let embeddedarticle = this.editor().getAttributes('embeddedarticle');

            let arguments = {
                type: 'embeddedarticle',
            };

            this.$dispatch('open-modal', { id: 'embedded-articles-modal', statePath: '{{ $statePath }}' }, arguments);
        }
    }">
    <svg xmlns="http://www.w3.org/2000/svg" width="20" height="20" viewBox="0 0 24 24">
        <path fill="currentColor"
            d="M5 21q-.825 0-1.413-.588T3 19V5q0-.825.588-1.413T5 3h14q.825 0 1.413.588T21 5v14q0 .825-.588 1.413T19 21H5Zm0-2h14v-5H5v5Z" />
    </svg>

    <span class="sr-only">asdas</span>
</x-filament-tiptap-editor::button>
```
**/my-app/resources/views/components/tiptap/extensions/embedded-articles/embedded-articles-modal.blade.php**

```blade
<x-filament::modal id="embedded-articles-modal" width="screen" class="embedded-articles-modal" displayClasses="block">
    <x-slot name="heading">
        Selecionar artigos
    </x-slot>
    <div>
        Modal for selecting articles
    </div>
</x-filament::modal>
```

**/my-app/app/TiptapExtensions/EmbeddedArticle.php**

```php

namespace App\TiptapExtensions;

use Tiptap\Core\Node;

class EmbeddedArticle extends Node
{
    public static $name = 'hero';
}
```
**/my-app/config/filament-tiptap-editor.php**

```php
    'extensions' => [
        [
            'id' => 'embeddedarticle',
            'name' => 'EmbeddedArticle',
            'button' => 'tiptap.extensions.embedded-articles.embedded-articles-button',
            'parser' => \App\TiptapExtensions\EmbeddedArticle::class,
        ],
    ],
```

**/my-app/resources/js/tiptap/EmbeddedArticle.js**

```javascript
import { Node } from '@tiptap/core'

export default Node.create({
    name: 'embeddedarticle',

    addNodeView () {
        return ({ editor, node, getPos, HTMLAttributes, decorations, extension }) => {
            const dom = document.createElement('div')

            dom.innerHTML = 'Hello, I’m a node view!'

            return {
                dom,
            }
        }
    },

    addCommands () {
        return {
            toggleEmbeddedArticle:
                () =>
                    ({ commands }) => {
                        return commands.toggleWrap(this.name);
                    },
        };
    },
})
```

**Comment by awcodes** [comment]
Curator's modal is outside of the form, that's what allows the panel to work inside of modals since the inserted image is passed via an event listener system. Because of this Curator has an even handler for the 'open-modal' event, that's why all the functionality inside that event handler is running a check for the modal id.

I think for your use case it would be better to just use a normal action instead of replicating what Curator is doing. It is a more complicated solution that what you need.

**Comment by MtDalPizzol** [comment]
Hey, @awcodes . I spent the last week trying to achieve this and it was really hard because of my lack of a deeper knowledge of internals from both Tiptap and Filament. But at end, I got where I wanted.

The thing is: I really tried using a normal action to open the modal when clicking the tool in the menu, but it resulted in A LOT  of weird interactions and unexpected behaviors, and I believe this is due to the fact that the action was inside the field itself.

The way I got everything working was just doing EXACTLY the same thing Curator does. Registering the modal view with a render hook, having a Livewire component for the modal with a "catch all" `open-modal` handler that checks for the modal id.
